service-discovery/consul/src/main/java/io/smallrye/stork/servicediscovery/consul/ConsulServiceDiscoveryProvider.java

```diff
@@ -13,14 +13,14 @@
 @ServiceDiscoveryAttribute(name = "use-health-checks", description = "Whether to use health check.", defaultValue = "true")
 @ServiceDiscoveryAttribute(name = "application", description = "The application name; if not defined Stork service name will be used.")
 @ServiceDiscoveryAttribute(name = "refresh-period", description = "Service discovery cache refresh period.", defaultValue = "5M")
+@ServiceDiscoveryAttribute(name = "secure", description = "whether the connection with the service should be encrypted with TLS.")
```

Comment:
shouldn't we assume it's always there and just generate code (and docs) for it regardless of the anntation being here?

Reply:
That was like that before , without the doc part.

The problem is that the semantic of this attribute depends on the protocol. For example, with Eureka it influences the endpoint selection. That's why it better to have it in each service discovery with an appropriate description.

Reply:
> The problem is that the semantic of this attribute depends on the protocol. For example, with Eureka it influences the endpoint selection. That's why it better to have it in each service discovery with an appropriate description.

Do you mean that the documentation may be different?

Reply:
do you think there are service discovery strategies that won't have this attribute?

Reply:
in the future, definitely. With TLS becoming the norm, it is likely that some service discovery providers will require that all the services use TLS. In this case, these service discovery providers will produce service instances with `secure=true`, but would not have the attribute, because it would be always true. They may reject instances using plain text.

When will we see this? It's hard to say. The main issue is that TLS is often seen as black magic and convoluted to set up. Once these technical difficulties are removed, it will be TLS everywhere. 

![image](https://user-images.githubusercontent.com/402301/156133384-31a37ef5-9336-4f1b-85d2-d9c398b7f1c8.png)